compiler/src/iree/compiler/Dialect/Stream/Transforms/test/specialize_encodings.mlir

```diff
@@ -182,6 +182,49 @@ module {
 
 // -----
 
+#executable_target_vmvx_bytecode_fb = #hal.executable.target<"vmvx", "vmvx-bytecode-fb", {encoding = #iree_encoding.unspecialized_encoding<123>}>
+#map = affine_map<(d0, d1, d2) -> (d0, d2)>
+#map1 = affine_map<(d0, d1, d2) -> (d2, d1)>
+#map2 = affine_map<(d0, d1, d2) -> (d0, d1)>
+#device_target_local_0_ = #hal.device.target<"local", {ordinal = 0 : index}, [#executable_target_vmvx_bytecode_fb]> : !hal.device
+#encoding = #iree_encoding.encoding<operand_index = 0 : index, op_type =  matmul, element_types = [f32, f32, f32], user_indexing_maps = [#map, #map1, #map2]>
```

Comment:
please use your test encodings

Reply:
The testing encoding attribute is used, see above "#iree_encoding.unspecialized_encoding<123>". This is the basic encodings for gemm.

Reply:
ah, then I think we need to either mark [all the parameters](https://github.com/iree-org/iree/blob/d81bb13bf695cfbd401af00c36889ce651fd2a17/compiler/src/iree/compiler/Dialect/Encoding/IR/EncodingAttrs.td#L77-L79) optional, or move the `cloneWithLayouts` to interface method and create a testing attribute for this.

Reply:
I'll do it in a follow-up. Do you have any preference? I think adding the testing attribute and the new interface method (e.g., `cloneWithEncoding`) makes sense. It helps us organize the needs for encoding specialization, i.e., only the encoding attributes that implement the interface can be specialized.